### PR TITLE
feat(tauri) add path API

### PR DESCRIPTION
**Pull request by lucasfernog**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] Feature
- [ ] New Binding Issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes. Issue #___
- [x] No

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by lucasfernog** [comment]
Works on Windows, Linux and macOS.
@nothingismagick need you to check the usage signature here, and let me know if you think we could improve it.

File: cli/tauri.js/api/fs/index.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  return tauri.readBinaryFile(filePath, options)
+}
+
+/**
+ * writes a text file
+ *
+ * @param {object} file
+ * @param {string} file.path path of the file
+ * @param {string} file.contents contents of the file
+ * @param {object} [options] configuration object
```
Comment (nothingismagick) [inline cli/tauri.js/api/fs/index.js]: Do we need an encoding? or do we assume UTF-8?

File: cli/tauri.js/api/fs/index.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  return tauri.writeFile(file, options)
+}
+
+/**
+ * @typedef {object} FileEntry
+ * @property {string} path
+ * @property {boolean} is_dir
+ * @property {string} name
+ */
+
```
Comment (nothingismagick) [inline cli/tauri.js/api/fs/index.js]: the above typedefs seem orphaned

File: cli/tauri.js/api/fs/index.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+ * @param {BaseDirectory} [options.dir] base directory
+ * @return {Promise<FileEntry[]>}
+ */
+function readDir (dir, options = {}) {
+  return tauri.readDir(dir, options)
+}
+
+/**
+ * Creates a directory
+ * If one of the path's parent components doesn't exist and the `recursive` option isn't set to true, it will be rejected
```
Comment (nothingismagick) [inline cli/tauri.js/api/fs/index.js]: Comment line is too long. Should wrap to next line.

File: tauri-api/src/path.rs [hunk 4]
```diff
@@ -0,0 +1,163 @@
+use std::path::PathBuf;
+use serde_repr::{Serialize_repr, Deserialize_repr};
+
+#[derive(Serialize_repr, Deserialize_repr, Clone, Debug)]
+#[repr(u16)]
+pub enum BaseDirectory {
+  Audio = 1,
```
Comment (nothingismagick) [inline tauri-api/src/path.rs]: Why is Audio the only one with a "Number"?

File: cli/tauri.js/api/fs/index.js (hunk 1 above)
Comment (lucasfernog) [reply cli/tauri.js/api/fs/index.js]: we just convert the string to a byte array, but we should probably have an endpoint to store binary files too

File: tauri-api/src/path.rs (hunk 4 above)
Comment (lucasfernog) [reply tauri-api/src/path.rs]: subsequent values increments from the first one

File: cli/tauri.js/api/fs/index.js (hunk 2 above)
Comment (lucasfernog) [reply cli/tauri.js/api/fs/index.js]: I use the FileEntry typedef on the readDir method

**Pull request merged**